### PR TITLE
solution

### PR DESCRIPTION
**Pull request by lefteris-L**
[DEMO LINK](https://lefteris-l.github.io/react_dynamic-list-of-todos/)

### CODE REVIEW

**Review by UmizDemud — APPROVED** [review]
Looks correct here 👍

**Review by ta1Fun1806 — CHANGES_REQUESTED** [review]
test

File: src/App.tsx [hunk 1]
```diff
@@ -17,18 +54,36 @@ export const App: React.FC = () => {
...
+              <TodoFilter
+                todos={todos}
+                setTodos={setVisibleTodos}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {!todos.length ? (
```
Comment (naziamoff) [inline src/App.tsx:64]: better to start with positive condition, it is easier to read
Also, the questionmark should be moved to a new line
```suggestion
              {!todos.length 
                ? (
```

File: src/components/TodoFilter/TodoFilter.tsx [hunk 2]
```diff
@@ -1,34 +1,97 @@
...
+      </p>
+
+      <p className="control is-expanded has-icons-left has-icons-right">
+        <input
+          data-cy="searchInput"
+          type="text"
+          value={query}
+          className="input"
+          placeholder="Search..."
+          onChange={(event) => {
```
Comment (naziamoff) [inline src/components/TodoFilter/TodoFilter.tsx:60]: looks way too huge, let's create a separate handler for that

File: src/components/TodoFilter/TodoFilter.tsx [hunk 3]
```diff
@@ -1,34 +1,97 @@
...
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={selected}
+            onChange={(event) => {
```
Comment (naziamoff) [inline src/components/TodoFilter/TodoFilter.tsx:39]: and here too

File: src/components/TodoList/TodoList.tsx [hunk 4]
```diff
@@ -1,100 +1,93 @@
...
-          <button data-cy="selectButton" className="button" type="button">
-            <span className="icon">
-              <i className="far fa-eye-slash" />
-            </span>
-          </button>
-        </td>
-      </tr>
+          </th>
+          <th>Title</th>
+          <th> </th>
```
Comment (naziamoff) [inline src/components/TodoList/TodoList.tsx:29]:
```suggestion
          <th />
```

File: src/components/TodoList/TodoList.tsx [hunk 5]
```diff
@@ -1,100 +1,93 @@
...
-              <i className="far fa-eye" />
-            </span>
-          </button>
-        </td>
-      </tr>
-    </tbody>
-  </table>
-);
+              <td className="is-vcentered is-expanded">
+                <p className={`has-text-${completed
```
Comment (naziamoff) [inline src/components/TodoList/TodoList.tsx:60]: classnames lib

File: src/components/TodoList/TodoList.tsx [hunk 6]
```diff
@@ -1,100 +1,93 @@
...
+                  className="button"
+                  type="button"
+                  onClick={() => {
+                    setTodo(todo);
+                    setUserId(userId);
+                  }}
+                >
+                  <span className="icon">
+                    <i
+                      className={`far fa-eye${selectedTodo.id === id
```
Comment (naziamoff) [inline src/components/TodoList/TodoList.tsx:80]: classnames lib

File: src/components/TodoModal/TodoModal.tsx [hunk 7]
```diff
@@ -1,49 +1,57 @@
...
+
+        <div className="modal-card-body">
+          <p className="block" data-cy="modal-title">
+            {todo.title}
+          </p>
+
+          <p className="block" data-cy="modal-user">
+            {todo.completed
+              ? <strong className="has-text-success">Done</strong>
+              : <strong className="has-text-danger">Planned</strong>}
```
Comment (naziamoff) [inline src/components/TodoModal/TodoModal.tsx:45]: can be done inside but one strong  + classnames library

**Review by naziamoff — CHANGES_REQUESTED** [review]
Not bad at all. I have a bunch of comments for making it even better

**Review by meforoma — APPROVED** [review]
👍🏼

**Pull request closed without merging**